modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/util/IntrospectionUtils.java

```diff
@@ -399,6 +402,44 @@ public static Object getFieldValue(Object object, String fieldName) throws Illeg
     }
   }
 
+  /**
+   * Retrieves the value of the parameter of name {@code parameterName} obtained from the {@code delegate} or
+   * from {@link ConfigurationInstance#getValue() config instance} if the {@code delegate} produces a
+   * {@code null} value.
+   *
+   * @param config the {@link ConfigurationInstance config} from where the parameter value will be obtained
+   * @param parameterName the name parameter to resolve
+   * @return the value of the parameter with name {@code parameterName} obtained from the {@code delegate} or
+   * from {@link ConfigurationInstance#getValue() config instance} if the {@code delegate} produces a {@code null} value.
+   */
+  public static Object resolveConfigOverrideParameter(Supplier delegate, Optional<ConfigurationInstance> config,
```

Comment:
Several things are wrong with this method:

* This doesn't belong here in the first place. IntrospectionUtils should at most have logic for locating the field/argument which is annotated, but going as far as actually evaluating and returning a value is not something that should happen here
* Fail fast. This performs a lot of checks which could fail but the supplier is evaluated before hand. You might incur into an expensive computation for no gain
* This method is being constantly called from the ValueResolver. This constant introspection of the model, etc should be done on the value resolver constructor and the field cached. Otherwise this is a bottleneck